crates/sncast/tests/integration/fee.rs

```diff
@@ -1,247 +1,238 @@
-    assert_eq!(u128::from(max_gas) * max_gas_unit_price, MAX_FEE.into());
 }
 
 #[tokio::test]
-async fn test_fee_get_max_gas_and_max_gas_unit_price() {
-    let factory = get_factory().await;
+async fn test_max_fee_set_and_fee_estimate_higher() {
+    let mock_fee_estimate = FeeEstimate {
+        l1_gas_consumed: Felt::from(10),
+        l1_data_gas_price: Felt::from(20),
+        l2_gas_consumed: Felt::from(30),
+        l2_gas_price: Felt::from(40),
+        l1_data_gas_consumed: Felt::from(50),
+        l1_gas_price: Felt::from(60),
+        unit: PriceUnit::Fri,
+        overall_fee: Felt::from(4400),
+    };
 
     let args = FeeArgs {
-        max_fee: None,
-        max_gas: Some(Felt::from(1_000_000_u32).try_into().unwrap()),
-        max_gas_unit_price: Some(Felt::from(1_000_u32).try_into().unwrap()),
+        max_fee: Some(NonZeroFelt::try_from(Felt::from(100)).unwrap()),
+        l1_gas: None,
+        l1_gas_price: None,
+        l2_gas: None,
+        l2_gas_price: None,
+        l1_data_gas: None,
+        l1_data_gas_price: None,
     };
 
-    let settings = args
-        .try_into_fee_settings(factory.provider(), factory.block_id())
-        .await
-        .unwrap();
+    let err = args
+        .try_into_fee_settings(Some(&mock_fee_estimate))
+        .unwrap_err();
 
     assert_eq!(
-        settings,
-        FeeSettings {
-            max_gas: Some(NonZeroU64::new(1_000_000).unwrap()),
-            max_gas_unit_price: Some(NonZeroU128::new(1_000).unwrap()),
-        }
+        err.to_string(),
+        format!(
+            "Estimated fee ({}) is higher than provided max fee ({})",
+            mock_fee_estimate.overall_fee,
+            Felt::from(args.max_fee.unwrap())
+        )
     );
 }
 
 #[tokio::test]
-async fn test_fee_get_max_fee_with_max_gas_unit_price() {
-    let factory = get_factory().await;
-
+#[should_panic(expected = "Fee estimate must be passed when max_fee is provided")]
+async fn test_max_fee_set_and_fee_estimate_none() {
     let args = FeeArgs {
-        max_fee: Some(Felt::from(MAX_FEE).try_into().unwrap()),
-        max_gas: None,
-        max_gas_unit_price: Some(Felt::from(1_000_u32).try_into().unwrap()),
+        max_fee: Some(NonZeroFelt::try_from(Felt::from(100)).unwrap()),
+        l1_gas: None,
+        l1_gas_price: None,
+        l2_gas: None,
+        l2_gas_price: None,
+        l1_data_gas: None,
+        l1_data_gas_price: None,
     };
 
-    let settings = args
-        .try_into_fee_settings(factory.provider(), factory.block_id())
-        .await
-        .unwrap();
+    let err = args.try_into_fee_settings(None).unwrap_err();
 
     assert_eq!(
-        settings,
-        FeeSettings {
-            max_gas: Some(NonZeroU64::new(MAX_FEE / 1_000).unwrap()),
-            max_gas_unit_price: Some(NonZeroU128::new(1_000).unwrap()),
-        }
+        err.to_string(),
+        "Fee estimate must be passed when max_fee is provided"
```

Comment:
I think it should just be 

```rust
args.try_into_fee_settings(None).unwrap();
```

Without any other lines? If the tests is `should_panic` anyway.